src/passes/MemoryCopyFillLowering.cpp

```diff
@@ -0,0 +1,230 @@
+          b.makeLocalSet(end, b.makeConst(-1U)),
+          b.makeLocalSet(step, b.makeConst(-1U)),
+        }),
+        b.makeBlock({
+          b.makeLocalSet(start, b.makeConst(0)),
+          b.makeLocalSet(end, b.makeLocalGet(size, Type::i32)),
+          b.makeLocalSet(step, b.makeConst(1)),
+        }))
+      );
+      // i = start
+      body->list.push_back(b.makeLocalSet(i, b.makeLocalGet(start, Type::i32)));
+      body->list.push_back(b.makeBlock(
+        "out",
+        b.makeLoop(
+          "copy",
+          b.makeBlock(
+            {// break if i == end
+             b.makeBreak("out",
+                         nullptr,
+                         b.makeBinary(BinaryOp::EqInt32,
+                                      b.makeLocalGet(i, Type::i32),
+                                      b.makeLocalGet(end, Type::i32))),
+             // dst[i] = src[i]
+             b.makeStore(1,
+                         0,
+                         1,
+                         b.makeBinary(BinaryOp::AddInt32,
+                           b.makeLocalGet(dst, Type::i32), b.makeLocalGet(i, Type::i32)),
+                         b.makeLoad(1,
+                                    false,
+                                    0,
+                                    1,
+                                    b.makeBinary(BinaryOp::AddInt32,
+                                      b.makeLocalGet(src, Type::i32), b.makeLocalGet(i, Type::i32)),
+                                    Type::i32,
+                                    memory),
+                         Type::i32,
+                         memory),
+              // i += step
+              b.makeLocalSet(i,
+                             b.makeBinary(BinaryOp::AddInt32,
+                                          b.makeLocalGet(i, Type::i32),
+                                          b.makeLocalGet(step, Type::i32))),
+              // loop
+              b.makeBreak("copy", nullptr)}))));
+      module->getFunction("__memory_copy")->body = body;
+
+      /*
+      local.set($temp, i32.mul(memory.size 0, i32.const 65536))
+      if (
+        i32.or(
+          i32.ugt(i32.add(local.get $src, local.get $size), local.get $temp),
+          i32.ugt(i32.add(local.get $dst, local.get $size), local.get $temp),
+        )
+      ) then unreachable
+      if (i32.lt(local.get $src), (local.get $dst)))
+      then (
+       local.set($start, i32.sub(local.get $size, i32.const 1)
+       local.set($end, (i32.const -1))
+       local.set($step, i32.const -1)
+       ) else (
+       local.set($start, i32.const 0)
+       local.set($end, local.get $size)
+       local.set($step, i32.const 1)
+      loop (
+       br_if (i32.eq(local.get $end, local.get $i)
+       i32.store8(i32.add((local.get $dst), (local.get $i)), 
+         i32.load8_u(i32.add((local.get $i) (local.get $src)))
+       local.set($dst, i32.add(local.get $i, (local.get $step)))
+      )
+      */
```

Comment:
I think these function creation block are large enough to warrant being in their own function, keeping `run` of reasonable length